saltboot/netutil.go

```diff
@@ -20,15 +35,35 @@ func DetermineBootstrapPort() int {
 	return port
 }
 
-func DetermineAuthCredentials() (string, string) {
-	username := os.Getenv("SALTBOOT_USERNAME")
-	password := os.Getenv("SALTBOOT_PASSWORD")
-	log.Printf("[determineAuthCredentials] SALTBOOT_USERNAME: %s SALTBOOT_PASSWORD: %s", username, password)
+func DetermineSecurityDetails(getEnv func (key string) string) (SecurityConfig, error) {
+	var (
+		password      string
+		signVerifyKey string
+	)
 
-	if len(strings.TrimSpace(username)) == 0 || len(strings.TrimSpace(password)) == 0 {
-		username = "cbadmin"
-		password = "cbadmin"
-		log.Printf("[determineAuthCredentials] using default credentials, username: %s, password: %s", username, password)
+	configLoc := strings.TrimSpace(getEnv(SALTBOOT_CONFIG))
+	if len(configLoc) == 0 {
+		password = strings.TrimSpace(getEnv(SALTBOOT_PASSWORD))
+		signVerifyKey = strings.TrimSpace(getEnv(SALTBOOT_SIGN_KEY))
+	} else {
+		remover := func (s string, r string) (string) {
+			return strings.TrimSpace(strings.Replace(s, r + "=", "", 1))
+		}
+		lines, _ := util.ReadLines(configLoc)
+		for i := range lines {
```

Comment:
I would prefer something like this: https://github.com/hortonworks/hdc-cli/blob/master/cli/config.go#L77

Basically the config file shall be yaml, just read it to SecurityConfig and if the env variable is set then overwrite it.